internal/utils/time_interval.go

```diff
@@ -0,0 +1,130 @@
+package utils
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+const (
+	// ErrEndBeforeStart is an the error message for when a TimeInterval's end time
+	// would be before its start.
+	ErrEndBeforeStart = "end time before start time"
+
+	errWindowTooLargeFmt = "random window equal to or larger than TimeInterval: window %v, interval %v"
+)
+
+// TimeInterval represents an interval of time in UTC. That is, regardless of
+// what timezone(s) are used for the beginning and end times, they will be
+// converted to UTC and methods will return them as such.
+type TimeInterval struct {
+	start time.Time
+	end   time.Time
+}
+
+// NewTimeInterval creates a new TimeInterval for a given start and end. If end
+// is a time.Time before start, then an error is returned.
+func NewTimeInterval(start, end time.Time) (*TimeInterval, error) {
+	if end.Before(start) {
+		return nil, fmt.Errorf(ErrEndBeforeStart)
```

Comment:
Readability nit: I would prefer to use `errors.New` here, even if it means a new import. `fmt.Errorf` uses that anyways and there is no formatting going on here.

Reply:
Hmm, I'm the opposite because I find often times I go back to error messages and add formatting options later. Even when I don't, I like the consistency of using or the other. I'm inclined to leave it like this until we do a more thorough errors refactoring to use best practices here:
https://dave.cheney.net/paste/gocon-spring-2016.pdf

Thoughts?

Reply:
That sounds good!